src/test/test.ts

```diff
@@ -145,14 +145,39 @@ export class Suite extends Base implements reporterTypes.Suite {
     return result;
   }
 
-  _hasOnly(): boolean {
+  _getOnlyItems(): (Spec | Suite)[] {
+    const items: (Spec | Suite)[] = [];
     if (this._only)
-      return true;
-    if (this.suites.find(suite => suite._hasOnly()))
-      return true;
-    if (this.specs.find(spec => spec._only))
-      return true;
-    return false;
+      items.push(this);
+    for (const suite of this.suites)
+      items.push(...suite._getOnlyItems());
+    items.push(...this.specs.filter(spec => spec._only));
+    return items;
+  }
+
+  _getUniqueItemsPerFile(): (Spec | Suite)[] {
+    const items: (Spec | Suite)[] = [];
+
+    function visit(suite: Suite, clashingSpecs: Record<string, Spec>, specToOccurence: Record<string, number>) {
+      for (const childSuite of suite.suites)
+        visit(childSuite, clashingSpecs, specToOccurence);
+      for (const spec of suite.specs) {
+        const fullTitle = spec.fullTitle();
+        if (!specToOccurence[fullTitle])
+          specToOccurence[fullTitle] = 0;
+        specToOccurence[fullTitle]++;
+        if (specToOccurence[fullTitle] > 1 && !clashingSpecs[fullTitle])
+          clashingSpecs[fullTitle] = spec;
```

Comment:
Hmm... As a user, I'd like to see all the occurrences, to figure out how to rename them properly. Otherwise, what should I do with a single one?